### PR TITLE
fix: devimint does not fetch channel funding txids for older gateways

### LINKED ISSUE #6077: Broken gateway upgrade tests
Gateway upgrade tests fail for `v0.4.* -> v0.4.3-rc.1`.

```
Error: Polling Open channel failed after 43 retries (timeout: 30s)
```

https://github.com/fedimint/fedimint/actions/runs/11005840625/job/30559138724#step:6:1433

You can reproduce locally with

```
TEST_KINDS=gateway just test-upgrades v0.4.0 v0.4.3-rc.1
```

I narrowed down the commit that introduced the breaking change to https://github.com/fedimint/fedimint/commit/c476322afeefb1702ed474e30463751c6c9d00b6. The commit doesn't revert cleanly so I'm currently running upgrade tests from the parent commit.

@tvolk131 I assigned your way since you have the most context :pray:

### PR DESCRIPTION
**Pull request by tvolk131**
Fixes #6077

The upgrade test was broken by #6008. This PR essentially reverts the behavior to what happened before that PR when running an old version of the gateway

### CODE REVIEW

**Comment by m1sterc001guy** [comment]
Concept ACK. Can you link to a successful upgrade test run so we can confirm it works?

**Review by bradleystachurski — APPROVED** [review]
@tvolk131 Thanks for tackling!

I verified these changes locally and in [CI](https://github.com/fedimint/fedimint/actions/runs/11026802219/job/30623996752) for `v0.4.0 -> v0.4.3-rc.1`

File: devimint/src/external.rs [hunk 1]
```diff
@@ -939,19 +939,31 @@ pub async fn open_channels_between_gateways(
...
+        } else {
+            is_missing_any_txids = true;
         }
     }
 
+    // If `open_channel` may not have sent out the channel funding transaction
+    // immediately. Since it didn't return a funding txid, we need to wait for
+    // it to get to the mempool.
+    if is_missing_any_txids {
+        fedimint_core::runtime::sleep(Duration::from_secs(5)).await;
```
[the remark this answers is not in the document]
Comment (dpc) [reply devimint/src/external.rs:964]: Any fixed delay in a test code is always a bug. Why can't we retry or something instead?

[the remark this answers is not in the document]
Comment (tvolk131) [reply devimint/src/external.rs]: There's nothing to retry. We'd need to just keep mining blocks and syncing the gateways while waiting for the until the channels to become ready. Would that be preferable?

**Review by dpc — APPROVED** [review]

**Review by bradleystachurski — APPROVED** [review]
Reran upgrade tests locally. Looks good :+1:

**Comment by tvolk131** [comment]
> Reran upgrade tests locally. Looks good :+1: 

Thanks Brad!

**Comment by elsirion** [comment]
Let's get tests green again!

**Pull request merged**